### PR TITLE
fix: client usage when `:check_settings` is `false`

### LINKED ISSUE #405: reindex with `check_settings: false`
- Rails version: 6.1.4
- Algolia Rails integration version: 2.1.2 (edit: the same with 2.1.3)
- Algolia Client Version: 2.1.1
- Language Version: 2.7.4

### Description
There is a problem in `.reindex` when using `check_settings: false`. I found that here: https://github.com/algolia/algoliasearch-rails/blob/master/lib/algoliasearch-rails.rb#L550 the object `@client` does not exist. I think that `AlgoliaSearch.client` has this method (`copy_index!`) but in my attempts to fix another error started to occur (`TypeError: no implicit conversion of Symbol into Integer`).

### Steps To Reproduce
Run `.reindex` with `algoliasearch(check_settings: false)`.

### PR DESCRIPTION
**Pull request by DevinCodes**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no    
| BC breaks?        | no     
| Related Issue     | Fix #405 

## Describe your change
This fixes a bug when doing a reindex while having `:check_settings` set to `false`.

### CODE REVIEW

**Comment by DevinCodes** [comment]
Hi @julienbourdeau , could you please review this PR if you have some spare time? Thank you in advance, let me know if you have any question! 🙇‍♂️

**Review by julienbourdeau — DISMISSED** [review]
Did you see the tests fail without your fix?

File: spec/integration_spec.rb [hunk 1]
```diff
@@ -162,6 +162,11 @@ def published?
 class Camera < Product
 end
 
+class Phone < Product
+  algolia_search :check_settings => false do
```
Comment (DevinCodes) [inline spec/integration_spec.rb:166]:
```suggestion
  algoliasearch :check_settings => false do
```

**Comment by DevinCodes** [comment]
> Did you see the tests fail without your fix?

They do fail without the fix, indeed! Sorry I spotted one minor issue, and I needed to update the branch. Can you re-approve, please? 🙂

**Review by tkrugg — APPROVED** [review]

**Pull request merged**